Force the default db directory to /var/lib/ldap, default to including
more than just core.schema, allow LDAPv2 clients, increase the set of
indexed attributes for the default database, and make it ldbm instead of
bdb because ldbm requires less attention, despite being both deprecated
and perhaps slower, and because back-bdb requires NPTL, which we don't
have everywhere yet.

--- openldap-2.1.17/doc/man/man8/slurpd.8	2003-03-03 12:10:03.000000000 -0500
+++ openldap-2.1.17/doc/man/man8/slurpd.8	2003-04-30 21:31:45.000000000 -0400
@@ -120,7 +120,7 @@
 temporary files may contain sensitive information.
 This option allows you to specify the location of these temporary files. 
 The default is
-.BR LOCALSTATEDIR/openldap-slurp .
+.BR /var/lib/ldap .
 .TP
 .BI \-k " srvtab\-file"
 Specify the location of the kerberos srvtab file which contains keys
--- openldap-2.1.17/servers/slapd/slapd.conf	2003-03-23 22:54:12.000000000 -0500
+++ openldap-2.1.17/servers/slapd/slapd.conf	2003-04-30 21:46:21.000000000 -0400
@@ -4,15 +4,20 @@
 # This file should NOT be world readable.
 #
 include		%SYSCONFDIR%/schema/core.schema
+include		%SYSCONFDIR%/schema/cosine.schema
+include		%SYSCONFDIR%/schema/inetorgperson.schema
+include		%SYSCONFDIR%/schema/nis.schema
+include		%SYSCONFDIR%/schema/redhat/autofs.schema
 
-# Define global ACLs to disable default read access.
+# Allow LDAPv2 client connections.  This is NOT the default.
+allow bind_v2
 
 # Do not enable referrals until AFTER you have a working directory
 # service AND an understanding of referrals.
 #referral	ldap://root.openldap.org
 
-pidfile		%LOCALSTATEDIR%/slapd.pid
-argsfile	%LOCALSTATEDIR%/slapd.args
+pidfile	/var/run/slapd.pid
+#argsfile	%LOCALSTATEDIR%/slapd.args
 
 # Load dynamic backend modules:
 # modulepath	%MODULEDIR%
@@ -22,6 +28,14 @@
 # moduleload	back_passwd.la
 # moduleload	back_shell.la
 
+# The next three lines allow use of TLS for connections using a dummy test
+# certificate, but you should generate a proper certificate by changing to
+# /usr/share/ssl/certs, running "make slapd.pem", and fixing permissions on
+# slapd.pem so that the ldap user or group can read it.
+# TLSCACertificateFile /usr/share/ssl/certs/ca-bundle.crt
+# TLSCertificateFile /usr/share/ssl/certs/slapd.pem
+# TLSCertificateKeyFile /usr/share/ssl/certs/slapd.pem
+
 # Sample security restrictions
 #
 #   Disallow clear text exchange of passwords
@@ -53,19 +67,32 @@
 # rootdn can always write!
 
 #######################################################################
-# ldbm database definitions
+# ldbm and/or bdb database definitions
 #######################################################################
 
-database	bdb
+database	ldbm
 suffix		"dc=my-domain,dc=com"
 rootdn		"cn=Manager,dc=my-domain,dc=com"
 # Cleartext passwords, especially for the rootdn, should
-# be avoid.  See slappasswd(8) and slapd.conf(5) for details.
+# be avoided.  See slappasswd(8) and slapd.conf(5) for details.
 # Use of strong authentication encouraged.
-rootpw		secret
+# rootpw		secret
+# rootpw		{crypt}ijFYNcSNctBYg
+
 # The database directory MUST exist prior to running slapd AND 
 # should only be accessible by the slapd and slap tools.
 # Mode 700 recommended.
-directory	%LOCALSTATEDIR%/openldap-data
-# Indices to maintain
-index	objectClass	eq
+directory	/var/lib/ldap
+
+# Indices to maintain for this database
+index objectClass                       eq,pres
+index ou,cn,mail,surname,givenname      eq,pres,sub
+index uidNumber,gidNumber,loginShell    eq,pres
+index uid,memberUid                     eq,pres,sub
+index nisMapName,nisMapEntry            eq,pres,sub
+
+# Replicas of this database
+#replogfile /var/lib/ldap/openldap-master-replog
+#replica host=ldap-1.example.com:389 tls=yes
+#     bindmethod=sasl saslmech=GSSAPI
+#     authcId=host/ldap-master.example.com@EXAMPLE.COM
--- openldap-2.1.17/servers/slurpd/slurp.h	2003-03-03 12:10:11.000000000 -0500
+++ openldap-2.1.17/servers/slurpd/slurp.h	2003-04-30 21:31:45.000000000 -0400
@@ -53,7 +53,7 @@
 #define SERVICE_NAME	OPENLDAP_PACKAGE "-slurpd"
 
 /* Default directory for slurpd's private copy of replication logs */
-#define	DEFAULT_SLURPD_REPLICA_DIR	LDAP_RUNDIR LDAP_DIRSEP "openldap-slurp"
+#define	DEFAULT_SLURPD_REPLICA_DIR	"/var/lib/ldap"
 
 /* Default name for slurpd's private copy of the replication log */
 #define	DEFAULT_SLURPD_REPLOGFILE	"slurpd.replog"
@@ -62,7 +62,7 @@
 #define	DEFAULT_SLURPD_STATUS_FILE	"slurpd.status"
 
 /* slurpd dump file - contents of rq struct are written here (debugging) */
-#define	SLURPD_DUMPFILE			LDAP_TMPDIR LDAP_DIRSEP "slurpd.dump"
+#define	SLURPD_DUMPFILE			DEFAULT_SLURPD_REPLICA_DIR "/slurpd.dump"
 
 /* default srvtab file.  Can be overridden */
 #define	SRVTAB				"/etc/srvtab"
